packages/client/src/app/pages/forms/edit/builder/tabs/layout/field-list/implementations/base-fields/modal/modal.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { SketchPicker } from 'react-color';
+import { Tooltip } from 'react-tippy';
+import { FormComponent } from '@components/form-controls';
+import { LoadingText } from '@components/loaders/loading-text/loading-text';
+import {
+  ModalContainer,
+  ModalFooter,
+  ModalHeader,
+} from '@components/modals/modal.styles';
+import type { ModalType } from '@components/modals/modal.types';
+import { useGroupMutation } from '@editor/builder/tabs/layout/property-editor/editors/fields/groups/groups.queries';
+import { useFetchGroups } from '@ff-client/queries/groups';
+import type { ErrorList } from '@ff-client/types/api';
+import type { FieldListRefs, Group } from '@ff-client/types/groups';
+import { PropertyType } from '@ff-client/types/properties';
+import translate from '@ff-client/utils/translations';
+import CrossIcon from '@ff-icons/actions/cross.svg';
+import MoveIcon from '@ff-icons/actions/move.svg';
+
+import {
+  CloseAndMoveWrapper,
+  ColorCircle,
+  ColorPicker,
+  ErrorBlock,
+  FieldListWrapper,
+  FieldTypes,
+  GroupHeader,
+  GroupItemWrapper,
+  GroupLayout,
+  GroupType,
+  GroupWrapper,
+  ManagerWrapper,
+  UHField,
+  UHFieldWrapper,
+} from './modal.styles';
+import { FieldItem } from './model.list-item';
+import { useGroupOperations } from './model.operations';
+import {
+  initializeGroupedSortable,
+  initializeSortable,
+} from './sortable.initializer';
+
+type ColorPickerProps = {
+  groupUid: string | null;
+  color: string;
+};
+
+export const CreateModal: ModalType = ({ closeModal }) => {
+  const [state, setState] = useState<Group>({});
+  const [colorPicker, setColorPicker] = useState<ColorPickerProps>();
+  const [errors, setErrors] = useState<ErrorList>();
+  const [loaded, setLoaded] = useState(false);
+
+  const fieldListRefs = useRef<FieldListRefs>({});
+  const { addGroup, updateGroupInfo, syncFromRefs } = useGroupOperations(
+    state,
+    setState,
+    fieldListRefs
+  );
+
+  const { data } = useFetchGroups();
+
+  useEffect(() => {
+    if (data && !loaded) {
+      setState(data);
+
+      setLoaded(true);
+    }
+  }, [data, loaded]);
```

Comment:
I'm not sure `loaded` is needed at all. It's only used to set the state.
We can do that by using the `isFetched` from the `useFetchGroups` query.

```suggestion
  useEffect(() => {
    if (isFetched) {
      setState(data);
    }
  }, [isFetched, data]);
```

Reply:
When 'Save' is clicked, the groups are saved to the database, and the modal closes. I don't want to reset the state;  It'll cause some problems.

Reply:
This shouldn't impact any thing.
* The state represents the initial dataset that comes from the endpoint.
* When you edit the groups, the state changes, then you click save - a mutation is triggered.
* Once the mutation is finished, the modal is closed and the query gets invalidated and re-fetched
* At this point, the modal is already closed, so it would not change the sate. But even if it did change the state, it would change it to the way it already is, because the state is persisted to the database.